src/core/shape_inference/include/squeeze_shape_inference.hpp

```diff
@@ -103,6 +117,43 @@ std::vector<TRShape> shape_infer(const Squeeze* op,
     } else {
         output_shape = PartialShape::dynamic();
     }
+}
+}  // namespace util
+namespace v0 {
+/**
+ * \brief Do Squeeze shape inference.
+ *
+ * \tparam T             Type of input/output shapes.
+ *
+ * \param op             Squeeze operator pointer.
+ * \param input_shapes   Squeeze input shapes.
+ * \param ta             Tensor accessor to constant data.
+ */
+template <class T, class TRShape = result_shape_t<T>>
+std::vector<TRShape> shape_infer(const Squeeze* op,
+                                 const std::vector<T>& input_shapes,
+                                 const ITensorAccessor& ta = make_tensor_accessor()) {
+    const auto& arg_shape = input_shapes[0];
+    const auto& arg_rank = arg_shape.rank();
+    auto output_shapes = std::vector<TRShape>(1);
+    auto& output_shape = output_shapes[0];
+    std::unique_ptr<std::set<int64_t>> unique_axes{};
+
+    auto output_shape_for_squeezable_dim = [&]() {
+        return PartialShape::dynamic(arg_rank.get_length() - 1);
+    };
+
+    if (ov::op::util::validate_input_and_try_set_output_shape(op,
```

Comment:
maybe this function should return optional shape and function `ov::op::util::set_output_shape` should return shape then:
```cpp
if (auto output_shape = ov::op::util::validate_input_and_try_set_output_shape(...)){
    output_shape.push_back(std::move(*output_shape));
} else{
    output_shape.push_back(ov::op::util::set_output_shape(...));
}


```

Reply:
Ok, good idea. Implemented.